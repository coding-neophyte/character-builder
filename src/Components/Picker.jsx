import React from 'react'


export default function Picker({ head, torso, legs, onHeadChange, onTorsoChange, onLegsChange, catchPhrase, addCatchPhrase, handleClick}) {

    const headOptions = ['bird', 'dog', 'duck']
    const torsoOptions = ['blue', 'fancy', 'pink', 'red']
    const legOptions = ['baggy', 'slim', 'loose', 'bootcut']



    return (
        <div>
            <label>
                Head
                <select value={head} onChange={(e) => onHeadChange(e.target.value)}>
                    {headOptions.map(headOp => <option key={headOp}>{headOp}</option>)}
                </select>
            </label>

            <label>
                Torso
                <select value={torso} onChange={(e) => onTorsoChange(e.target.value)}>
                    {torsoOptions.map(middle => <option key={middle}>{middle}</option>)}
                </select>
            </label>

            <label>
                Legs
                <select value={legs} onChange={(e) => onLegsChange(e.target.value)}>
                    {legOptions.map(leg => <option key={leg}>{leg}</option>)}
                </select>
            </label>
            <label>
                Add A Catchphrase
                <input value={catchPhrase} onChange={(e) => addCatchPhrase(e.target.value)}></input>
                <button type='button' onClick={() => handleClick()}> Add </button>
            </label>
        </div>
    )
}
