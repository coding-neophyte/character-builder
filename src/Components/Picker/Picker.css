.button{
    height: 25px;
    width: 60px;
    margin-top: 13px;
    color: white;
    background-color: blue;
    border-radius: 8px;
    border-color: lightblue;
}
