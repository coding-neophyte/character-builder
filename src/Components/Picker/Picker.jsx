import React from 'react'
import './Picker.css'

export default function Picker({
    head,
    torso,
    legs,
    onHeadChange,
    onTorsoChange,
    onLegsChange,
    catchPhrase,
    setCatchPhrase,
    handleClick}) {

    const headOptions = ['cap', 'beanie', 'tophat', 'bucket']
    const torsoOptions = ['polo', 'sweater', 'buttonup', 'cardigan']
    const legOptions = ['baggy', 'slimfit', 'skinny', 'bootcut']



    return (
        <div style={{display: 'flex', justifyContent: 'center'}}>
            <label>
                Head
                <select value={head} onChange={(e) => onHeadChange(e.target.value)}>
                    {headOptions.map(headOp => <option key={headOp}>{headOp}</option>)}
                </select>
            </label>

            <label>
                Torso
                <select value={torso} onChange={(e) => onTorsoChange(e.target.value)}>
                    {torsoOptions.map(middle => <option key={middle}>{middle}</option>)}
                </select>
            </label>

            <label>
                Legs
                <select value={legs} onChange={(e) => onLegsChange(e.target.value)}>
                    {legOptions.map(leg => <option key={leg}>{leg}</option>)}
                </select>
            </label>
            <label>
                Add A Catchphrase
                <input type='text' value={catchPhrase} onChange={(e) => setCatchPhrase(e.target.value)}></input>
            </label>
            <button type='button' onClick={() => handleClick()} className='button'> Add </button>
        </div>
    )
}
