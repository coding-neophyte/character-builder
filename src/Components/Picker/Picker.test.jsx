import { screen, render } from '@testing-library/react'
import Picker from './Picker'

it ('should render picker component', () => {
    const {container} = render(<Picker />)
    const selectHead = screen.getByLabelText(/Head/);
    const selectTorso = screen.getByLabelText(/Torso/);
    const selectLegs = screen.getByLabelText(/Legs/);
    const input = screen.getByLabelText(/Add A Catchphrase/)


    expect(selectHead).toBeInTheDocument();
    expect(selectHead).toBeInTheDocument();
    expect(selectTorso).toBeInTheDocument();
    expect(selectLegs).toBeInTheDocument();
    expect(input).toBeInTheDocument();
    expect(container).toMatchSnapshot();
});
