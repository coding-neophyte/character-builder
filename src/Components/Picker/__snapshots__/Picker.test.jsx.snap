// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render picker component 1`] = `
<div>
  <div
    style="display: flex; justify-content: center;"
  >
    <label>
      Head
      <select>
        <option>
          cap
        </option>
        <option>
          beanie
        </option>
        <option>
          tophat
        </option>
        <option>
          bucket
        </option>
      </select>
    </label>
    <label>
      Torso
      <select>
        <option>
          polo
        </option>
        <option>
          sweater
        </option>
        <option>
          buttonup
        </option>
        <option>
          cardigan
        </option>
      </select>
    </label>
    <label>
      Legs
      <select>
        <option>
          baggy
        </option>
        <option>
          slimfit
        </option>
        <option>
          skinny
        </option>
        <option>
          bootcut
        </option>
      </select>
    </label>
    <label>
      Add A Catchphrase
      <input
        type="text"
        value=""
      />
    </label>
    <button
      class="button"
      type="button"
    >
       Add 
    </button>
  </div>
</div>
`;
